### ISSUE TITLE
UnhandledPromiseRejectionWarning

### ISSUE THREAD

**Issue by susan8098** [11ty/eleventy#110]
I am just getting started, and running 'eleventy' kept giving me this error:
```
(node:1800) UnhandledPromiseRejectionWarning: #<EleventyError>
(node:1800) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 1)
(node:1800) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```
What are some things I can try?

**Comment by zachleat** [comment]
Oh! Well, that’s no good! Did you try debug mode? https://github.com/11ty/eleventy/#debugging

Run `DEBUG=Eleventy* eleventy` and it’ll spit out a bunch of stuff you can add here!

What’s your directory structure / `.eleventy.js` config file look like?

**Comment by zachleat** [comment]
No matter what’s happening here we need better error handling and messaging.

**Comment by chrisdmacrae** [comment]
My thoughts on this are to throw errors everywhere and catch them all in the root index file.

**Comment by susan8098** [comment]
I have been trying a few different things. Right now this is what my `.eleventy.js` looks like, and it's working fine right now:
```
module.exports = {
  dir: {
    input: "src",
    data: "_data"
  },
  templateFormats: ['pug','md', 'css', 'eot', 'svg', 'ttf', 'woff', 'png'],
};
```

When I pass in `eleventyConfig` like this though:
```
module.exports = function(eleventyConfig) {
  return {
    templateFormats: ['pug','md', 'css', 'eot', 'svg', 'ttf', 'woff', 'png'],
    dir: {
      input: ".",
      includes: "_includes",
      data: "_data",
      output: "_site"
    }C
  };
};
```
it will then give me an error:
```
Error: TemplateLayout directory does not exist for layout.pug: ./_includes
```
My directory structure looks like
```
/project
    /src
        /_data
        /_includes
            /layout.pug
        /assets
        /css
        index.pug
    eleventy.js
```

**Comment by zachleat** [comment]
```
module.exports = function(eleventyConfig) {
  return {
    templateFormats: ['pug','md', 'css', 'eot', 'svg', 'ttf', 'woff', 'png'],
    dir: {
      input: ".",
      includes: "_includes",
      data: "_data",
      output: "_site"
    }C
  };
};
```
I’m gonna assume that C is not there on your local?

Regardless, I don’t think the `dir`’s in your second example match the first? I think you want this:

```
module.exports = function(eleventyConfig) {
  return {
    templateFormats: ['pug','md', 'css', 'eot', 'svg', 'ttf', 'woff', 'png'],
    dir: {
      input: "src",
    }
  };
};
```

All the others are default and don’t need overrides. This will resolve to:

* Input directory: `"src"`
* Includes directory: `"src/_includes"`
* Data directory: `"src/_data"`
* Output directory: `"_site"`

Does that make sense?

**Comment by zachleat** [comment]
Another thing that might simplify it for you is to use a directory passthrough copy instead of all of those template formats:

```
module.exports = function(eleventyConfig) {
  eleventyConfig.addPassthroughCopy("src/assets");
  eleventyConfig.addPassthroughCopy("src/css");

  return {
    templateFormats: ['pug','md'],
    dir: {
      input: "src",
    }
  };
};
```

**Comment by susan8098** [comment]
ahh right, nice spot, and I will take note of passthrough. 
Thank you so much for the help!

**Comment by paulrobertlloyd** [comment]
I’ve started seeing this error too (although I don’t know if this was happening before and I missed it, or something has changed to have it start happening). Here is the error, within the context of the debug output:

```
  Dev:Eleventy:Template new Template('./src/_indices/archives.liquid') +0ms
  Dev:Eleventy:Template './src/_indices/archives.liquid' getMapped() +5ms
  Dev:Eleventy:Template './src/_indices/archives.liquid' getData() +1ms
  Dev:Eleventy:TemplateData getLocalDataPaths('./src/_indices/archives.liquid') +0ms
  Dev:Eleventy:TemplateData parsed.dir: './src/_indices' +0ms
  Dev:Eleventy:TemplateData allDirs: [ './src/_indices', './src' ] +0ms
  Dev:Eleventy:TemplateData dirStr: './src/_indices'; inputDir: './src' +0ms
  Dev:Eleventy:TemplateData dirStr: './src'; inputDir: './src' +0ms
  Eleventy:TemplateData getLocalDataPaths('./src/_indices/archives.liquid'): [ './src/_indices/archives.json', './src/_indices/_indices.json' ] +135ms
(node:42087) UnhandledPromiseRejectionWarning: #<EleventyError>
(node:42087) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 1)
(node:42087) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```

**Comment by paulrobertlloyd** [comment]
Ah ha! I’ve found out what I did! I changed the name of my assets folder from `assets` to `_assets`, but hadn’t updated this line in my 11ty config:

```
eleventyConfig.addPassthroughCopy('src/assets');
```

Correcting the folder name meant this error went away.

**Comment by zachleat** [comment]
Looks like this will improve with some of the changes in #112.

**Comment by zachleat** [comment]
Okay, this will be included with The Next Minor Version after v0.3.5
